spec/lib/hackney/income/stored_tenancies_gateway_spec.rb

```diff
@@ -416,6 +416,94 @@
     end
   end
 
+  context 'when there are tenancies with different patches' do
+    let(:patch_1) { Faker::Lorem.characters(3) }
+    let(:patch_2) { Faker::Lorem.characters(3) }
+
+    let(:user) { create(:user) }
+
+    let(:num_cases_in_patch_1) { Faker::Number.between(2, 10) }
+    let(:num_cases_in_patch_2) { Faker::Number.between(2, 20) }
+    let(:num_pages) { Faker::Number.between(1, 5) }
+
+    before do
+      num_cases_in_patch_1.times do
+        create(:case_priority, assigned_user_id: user.id, balance: 40, patch_code: patch_1)
+      end
+
+      num_cases_in_patch_2.times do
+        create(:case_priority, assigned_user_id: user.id, balance: 40, patch_code: patch_2)
+      end
+    end
+
+    context 'when we call get_tenancies_for_user' do
+      subject do
+        gateway.get_tenancies_for_user(
+          user_id: user.id,
+          page_number: 1,
+          number_per_page: 50,
+          patch: patch
+        )
+      end
+
+      let(:patch) { nil }
+
+      it 'returns all tenancies' do
+        expect(subject.count).to eq(num_cases_in_patch_1 + num_cases_in_patch_2)
+      end
+
+      context 'when and filtering by patch 1' do
+        let(:patch) { patch_1 }
+
+        it 'onlies return only paused tenancies' do
+          expect(subject.count).to eq(num_cases_in_patch_1)
+        end
+      end
+
+      context 'when and filtering by patch 2' do
+        let(:patch) { patch_2 }
+
+        it 'onlies return unpaused tenancies' do
+          expect(subject.count).to eq(num_cases_in_patch_2)
+        end
+      end
+    end
+
+    context 'when we call number_of_pages_for_user' do
```

Comment:
I'm happy using the method name convention for this level of spec, but don't feel strongly
```suggestion
    context '#number_of_pages_for_user' do
```